core/chains/evm/gas/models.go

```diff
@@ -164,6 +168,26 @@ func (e WrappedEvmEstimator) GetFee(ctx context.Context, calldata []byte, feeLim
 	return
 }
 
+func (e WrappedEvmEstimator) GetCost(ctx context.Context, amount assets.Eth, calldata []byte, feeLimit uint32, maxFeePrice *assets.Wei, opts ...feetypes.Opt) (*big.Int, error) {
+	fees, gasLimit, err := e.GetFee(ctx, calldata, feeLimit, maxFeePrice, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	var gasPrice *assets.Wei
+	if e.EIP1559Enabled {
+		gasPrice = fees.DynamicFeeCap
```

Comment:
Note that this is not the amount that will be spent on gas, it's the maximum amount that could be spent on gas.

It's not useful for calculations on gas that might drain the wallet.

Reply:
this is true!

just noting our discussion: it's a bit of a tangential issue. in this case the the estimation is used to validate so it's possible that it could fail more frequently depending on the estimation

however, precisely draining a wallet would still be difficult because the tx sent through the transaction manager would still leverage an estimated fee value - so it's possible it may not drain the wallet regardless of validation